src/main/scala/org/apache/mesos/chronos/scheduler/api/JobManagementResource.scala

```diff
@@ -142,14 +142,21 @@ class JobManagementResource @Inject()(val jobScheduler: JobScheduler,
   @Path(PathConstants.jobPatternPath)
   @PUT
   @Timed
-  def trigger(@PathParam("jobName") jobName: String,
-              @QueryParam("arguments") arguments: String
-               ): Response = {
+  def trigger(@PathParam("jobName") jobName: String, jobParams: JobParams): Response = {
     try {
       require(jobGraph.lookupVertex(jobName).isDefined, "Job '%s' not found".format(jobName))
       val job = jobGraph.getJobForName(jobName).get
       log.info("Manually triggering job:" + jobName)
-      jobScheduler.taskManager.enqueue(TaskUtils.getTaskId(job, DateTime.now(DateTimeZone.UTC), 0), job.highPriority)
+
+      var extraJobArguments: Seq[String] = List()
+
+      if (jobParams!=null) {
```

Comment:
Using `Option(jobParams).map(_.arguments).getOrElse(List.empty)` would be more idiomatic, in my opinion.